iis-wf/iis-wf-import/src/test/java/eu/dnetlib/iis/wf/importer/infospace/converter/MetadataConverterUtilsTest.java

```diff
@@ -0,0 +1,106 @@
+package eu.dnetlib.iis.wf.importer.infospace.converter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import java.time.Year;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.dhp.schema.oaf.DataInfo;
+import eu.dnetlib.dhp.schema.oaf.StructuredProperty;
+import eu.dnetlib.iis.wf.importer.infospace.approver.FieldApprover;
+
+/**
+ * {@link MetadataConverterUtils} test class. 
+ * 
+ * @author mhorst
+ *
+ */
+public class MetadataConverterUtilsTest {
+
+    private Logger log = mock(Logger.class);
+    
+    @Test
+    public void testExtractYearOrNullWithValidInput() throws Exception {
+        assertEquals(Year.of(2020), MetadataConverterUtils.extractYearOrNull("2020-12-31", log));
+    }
+    
+    @Test
+    public void testExtractYearOrNullWithInvalidInput() throws Exception {
+        assertNull(MetadataConverterUtils.extractYearOrNull("invalid", log));
+        assertNull(MetadataConverterUtils.extractYearOrNull("20-20-01", log));
+        assertNull(MetadataConverterUtils.extractYearOrNull("2020-01", log));
+        assertNull(MetadataConverterUtils.extractYearOrNull("2020-01-1", log));
+        assertNull(MetadataConverterUtils.extractYearOrNull("", log));
+        assertNull(MetadataConverterUtils.extractYearOrNull(null, log));
+    }
+    
+    @Test(expected = NullPointerException.class)
+    public void testExtractValuesForNullApprover() throws Exception {
+        // given
+        Collection<StructuredProperty> source = Lists.newArrayList();
+        String validValue = "someValue";
+        source.add(generateStructuredProperty(validValue, null));
+        
+        // execute
+        MetadataConverterUtils.extractValues(source, null);
+    }
+    
+    @Test(expected = NullPointerException.class)
+    public void testExtractValuesForNullSource() throws Exception {
+        MetadataConverterUtils.extractValues(null, mock(FieldApprover.class));
+    }
+    
+    @Test
+    public void testExtractValuesForEmptySource() throws Exception {
+        //execute
+        List<String> results = MetadataConverterUtils.extractValues(Lists.newArrayList(), mock(FieldApprover.class));
+        
+        //assert
+        assertNotNull(results);
+        assertTrue(results.isEmpty());
+    }
+    
+    @Test
+    public void testExtractValues() throws Exception {
+        // given
+        @SuppressWarnings("serial")
+        FieldApprover notNullDataInfoFieldApprover = new FieldApprover() {
+            @Override
+            public boolean approve(DataInfo dataInfo) {
+                return dataInfo != null;
+            }
+        };
```

Comment:
IDE hint: this can be substituted with `FieldApprover notNullDataInfoFieldApprover = Objects::nonNull;`

Reply:
This is quite nice. 

At first glance the level of abstraction here could be overwhelming, at least until realizing functional interfaces or SAM classes can be treated in a very special way since java8.

I need to enable those neat little hints in my IDE.

Reply:
Yes, it is quite nice and concise. I generally start with explicit anonymous class definition and follow IDE hints to replace any code parts with more abstract versions.